### PR TITLE
Activity overview page

### PR DESCRIPTION
**Pull request by DutcherNL**
Added a visual activity display page instead of the calendar
Moved calendar to different page
Added short instructions on importing calendar in your own calendar

### CODE REVIEW

File: activity_calendar/models.py [hunk 1]
```diff
@@ -56,6 +56,7 @@ class Meta:
     description = MarkdownTextField(help_text="Note that uploaded images are publicly accessible, even if the activity is unpublished.")
     location = models.CharField(max_length=255)
     image = models.ForeignKey(PresetImage, blank=True, null=True, related_name="activity_image", on_delete=models.SET_NULL)
```
Comment (EricTRL) [inline activity_calendar/models.py:58]: You could rename `image` to `slots_image`. Either way, adding in some help_text would be nice

File: activity_calendar/templates/activity_calendar/activity_moment_form_page.html [hunk 2]
```diff
@@ -91,7 +91,7 @@ <h4>Edit occurence data</h4>
         it will copy the attribute from the general activity instead.
     </p>
 
-    {% render_form has_submit_button=False id="submit_form" method="post" %}
+    {% render_form has_submit_button=False id="submit_form" method="post" uses_file_field=True %}
```
Comment (EricTRL) [inline activity_calendar/templates/activity_calendar/activity_moment_form_page.html:94]: Clearable file-inputs don't really work with the current form layout

<details><summary>Screenshot</summary>

image
</details>

File: activity_calendar/templates/activity_calendar/activity_overview.html [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            </div>
+        {% endfor %}
+        </div>
+
+    </div>
+{% endblock %}
+
+
+
+{% block css %}
```
Comment (EricTRL) [inline activity_calendar/templates/activity_calendar/activity_overview.html:36]: I don't think we need Martor's CSS on this page; the activity descriptions aren't shown here.

File: activity_calendar/templates/activity_calendar/fullcalendar.html [hunk 4]
```diff
@@ -52,120 +52,132 @@ <h1>Activity Calendar</h1>
...
-                    <p id="event-date"></p>
-                    <p><small id="occurrence-replacement" class="text-muted"></small></p>
-                    <span id="event-recurrence-info">
-                      <p id="rrules"></p>
-                      <p id="rdates"></p>
-                    </span>
-                  </small>
-                </div>
-              </div>
+    <div class="modal fade" id="event-modal" tabindex="-1" role="dialog" aria-labelledby="event-modal-aria" aria-hidden="true">
```
Comment (EricTRL) [inline activity_calendar/templates/activity_calendar/fullcalendar.html:55]: I think it's just the indentation that changed here right?

File: activity_calendar/templates/activity_calendar/activity_overview.html [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    Squire - Boardgames
+{% endblock title %}
+
+{% block content-frame-class %}
+    container-fluid
+{% endblock %}
+
+{% block content %}
+    <div class="m-auto" style="max-width: 1250px;">
+        <h1>Activities</h1>
```
Comment (EricTRL) [inline activity_calendar/templates/activity_calendar/activity_overview.html:15]: I suggest renaming the title to "Upcoming Activities", and to add an "All Activities"-button as I feel like this link is a bit too hidden.

File: activity_calendar/templates/activity_calendar/fullcalendar.html [hunk 6]
```diff
@@ -52,120 +52,132 @@ <h1>Activity Calendar</h1>
...
         </div>
-        <div class="modal-footer">
-          <a id='event-subscribe-url' class="btn btn-primary" href="">View Registration Slots</a>
-          <button type="button" class="btn btn-secondary" data-dismiss="modal">Close</button>
-        </div>
-      </div>
     </div>
-  </div>
+
+    <div class="mt-3">
```
Comment (EricTRL) [inline activity_calendar/templates/activity_calendar/fullcalendar.html:172]: This takes up a lot of space for what is essentially something that people do at most once. Possibly move it to a modal that opens on a button click?

File: activity_calendar/templates/activity_calendar/fullcalendar.html [hunk 7]
```diff
@@ -52,120 +52,132 @@ <h1>Activity Calendar</h1>
...
+        <h5>Add this to your agenda</h5>
+        <p>You can import the Knights agenda into your own agenda. To do this follow the instructions on adding an
+        agenda from url and use the following url:</p>
+        <p class="">
+            https://app.kotkt.nl{% url "activity_calendar:icalendar" %}
```
Comment (EricTRL) [inline activity_calendar/templates/activity_calendar/fullcalendar.html:177]: Hardcoded URLs are bound to break at some point. Perhaps look into the `build_absolute_uri` template tag (in `core`); it _almost_ works for this case.

File: activity_calendar/templates/activity_calendar/snippet_activity_moment.html [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+                </div>
+                <!-- Subscription status -->
+                <div class="col align-self-center">
+                    <p id="event-status">
+                        {% if user|is_subscribed_to:activity_moment %}
+                            <span class="text-success">
+                                You are subscribed
+                            </span>
+                        {% elif activity_moment.is_open_for_subscriptions %}
+                            {% if activity_moment.get_subscribed_users.count >= activity_moment.max_participants and activity_moment.max_participants != -1 %}
```
Comment (EricTRL) [inline activity_calendar/templates/activity_calendar/snippet_activity_moment.html:52]: At some point we might want to make an `ActivityMoment.is_full(..)` method; we're doing this same check in more places

File: activity_calendar/templates/activity_calendar/snippet_activity_moment.html [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+                        {% endif %}
+                    </p>
+                </div>
+            </div>
+
+            <!-- Link to activity moment page -->
+
+
+            <a href="{{ activity_moment.get_absolute_url }}" class="card-link btn btn-primary"
+                style="z-index: -1000;">
```
Comment (EricTRL) [inline activity_calendar/templates/activity_calendar/snippet_activity_moment.html:79]: Any particular reason for this `z-index`?

File: activity_calendar/templatetags/calendar_tags.py [hunk 10]
```diff
@@ -39,6 +39,11 @@ def get_opening_time(context, filter=None):
     return open_date
 
 
+@register.filter
+def is_subscribed_to(user, activity_moment):
+    return activity_moment.get_user_subscriptions(user)
```
Comment (EricTRL) [inline activity_calendar/templatetags/calendar_tags.py:44]: Judging from the method name, shouldn't this return a boolean?

**Review by EricTRL — CHANGES_REQUESTED** [review]
Overall I like the changes, but I have a few points of feedback/food for thought:
1) Wouldn't the "upcomign activities' fit on the homepage itself? It''ll likely be the most often used feature. The calendar can then be readded to the navbar, or the "activities"-tab can be removed alltogether.
2) We have a few important activities each year that are planned way further ahead. Think of Fantasy Court, Luna Fest, the long boardgame days, Spiel, the LFG, or even the board's constitution drink. In the calendar they're usually visible earlier (due to it defaulting to the month view), but right now they'll remain sort of hidden until they're almost there. Perhaps you could only show activities whose subscriptions are open? Alternatively, add a new field to the Activity model to indicate when activities will show up (defaulting to "two weeks in advance")?
3) Is it really needed to have multiple occurrences of the same activity show up? There's not really a point in having two boardgame evening's in there at once.

File: activity_calendar/templates/activity_calendar/fullcalendar.html (hunk 6 above)
Comment (DutcherNL) [reply activity_calendar/templates/activity_calendar/fullcalendar.html:172]: One day in the future (r) I made an issue for it #148

File: activity_calendar/templatetags/calendar_tags.py (hunk 10 above)
Comment (DutcherNL) [reply activity_calendar/templatetags/calendar_tags.py:44]: Technically yes, this hasn't caught my attention yet as the template engine interprets an empty iterable as False XD

**Review by EricTRL — APPROVED** [review]

**Pull request merged**